src/deepsparse/v2/pipeline.py

```diff
@@ -136,17 +135,13 @@ def _run_sub_graphs(
                     # update the output value
                     if next_step in sub_graph.end:
                         sub_graph.output = operator_output
+                        sub_graph.completed = True
                     else:
                         sub_graph.output = self._run_next(
                             inp=operator_output,
                             inference_state=sub_graph.inf,
                             next_step=next_step,
                         )
-                    break
-
-            # keep running until all sub graphs have completed.
-            if not any(isinstance(x.output, Future) for x in sub_graphs):
```

Comment:
So what was the core issue? We were waiting idly for all the subgraphs?

Reply:
From what I understand, I was using the concurrent Future `done()` method incorrectly.